src/execution/index/art/art.cpp

```diff
@@ -33,15 +30,16 @@ struct ARTIndexScanState : public IndexScanState {
 	Iterator iterator;
 };
 
-ART::ART(const vector<column_t> &column_ids, TableIOManager &table_io_manager,
-         const vector<unique_ptr<Expression>> &unbound_expressions, const IndexConstraintType constraint_type,
+//===--------------------------------------------------------------------===//
+// ART
+//===--------------------------------------------------------------------===//
+
+ART::ART(const string &name, const IndexConstraintType index_constraint_type, const vector<column_t> &column_ids,
+         TableIOManager &table_io_manager, const vector<unique_ptr<Expression>> &unbound_expressions,
          AttachedDatabase &db, const shared_ptr<array<unique_ptr<FixedSizeAllocator>, ALLOCATOR_COUNT>> &allocators_ptr,
-         const BlockPointer &pointer)
-    : Index(db, IndexType::ART, table_io_manager, column_ids, unbound_expressions, constraint_type),
+         const IndexStorageInfo &index_storage_info)
+    : Index(name, "ART", index_constraint_type, column_ids, table_io_manager, unbound_expressions, db),
```

Comment:
So, the index type here is changed to allow for pluggable indexes. 
I'm wondering if it makes sense to have a dynamic enum here (i.e., a static enum with an extra map for indexes created in extensions) to avoid error-prone string comparisons switcharoo.
I think these switcharoos will have to happen in the core, since theoretically, you can have multiple extensions with different index structures that must co-exist. But maybe we can have a bigger discussion on the pluggable indexes strategy later. cc @Maxxen